Packs/Absolute/Integrations/Absolute/Absolute.py

```diff
@@ -0,0 +1,947 @@
+    else:
+        query = add_list_to_filter_string("deviceName", device_names, query)
+
+    app_names = remove_duplicates_from_list_arg(args, 'app_names')
+    query = add_list_to_filter_string("appName", app_names, query)
+
+    app_publishers = remove_duplicates_from_list_arg(args, 'app_publishers')
+    query = add_list_to_filter_string("appPublisher", app_publishers, query)
+
+    user_names = remove_duplicates_from_list_arg(args, 'user_names')
+    query = add_list_to_filter_string("userName", user_names, query)
+
+    operating_systems = remove_duplicates_from_list_arg(args, 'os')
+    query = add_list_to_filter_string("osName", operating_systems, query)
+
+    devices_esn = remove_duplicates_from_list_arg(args, 'esn')
+    query = add_list_to_filter_string("esn", devices_esn, query)
+
+    local_ips = remove_duplicates_from_list_arg(args, 'local_ips')
+    query = add_list_to_filter_string("localIp", local_ips, query)
+
+    public_ips = remove_duplicates_from_list_arg(args, 'public_ips')
+    query = add_list_to_filter_string("publicIp", public_ips, query)
+
+    if args.get('agent_status'):
+        agent_status = ABSOLUTE_AGET_STATUS[args.get('agent_status')]  # type: ignore
+        query = add_value_to_filter_string("agentStatus", agent_status, query)
+
+    os_name = args.get('os_name')
+    query = add_value_to_filter_string("osName", os_name, query)
+
+    os_version = args.get('os_version')
+    query = add_value_to_filter_string("osVersion", os_version, query)
+
+    manufacturer = args.get('manufacturer')
+    query = add_value_to_filter_string("systemManufacturer", manufacturer, query)
+
+    model = args.get('model')
+    query = add_value_to_filter_string("systemModel", model, query)
+
+    return f"$filter={query}"
+
+
+def parse_return_fields(return_fields: str, query: str):
+    """
+    Returns values only for the fields that meet the specified criteria in the query.
+    All other fields are returned with a null value.
+    """
+    if not return_fields:
+        return query
+
+    if query:
+        return f"{query}&$select={return_fields}"
+    return f"$select={return_fields}"
+
+
+def parse_paging(page: int, limit: int, query: str) -> str:
+    """
+    Add pagination query format to the existing query
+    """
+    if query:
+        return f'{query}&$skip={page}&$top={limit}'
+    return f"$skip={page}&$top={limit}"
+
+
+def parse_device_list_response(response, keep_os_in_list=True):
+    parsed_response = []
+    for device in response:
+        parsed_device = {}
+        for key, val in device.items():
+            if val:
+                if key == 'os' and not keep_os_in_list:
+                    parsed_device['osName'] = val.get('name')
+                elif key == 'espInfo':
+                    parsed_device['encryptionStatus'] = val.get('encryptionStatus')
+                else:
+                    parsed_device[key[0].upper() + key[1:]] = val
```

Comment:
Please add a comment here about the expected type of `key` - how come can this be a list?

Reply:
Key is not a list, is a string and we take the substring of the first place till the end. The key is actually the devices fields.